docs/multi-pp-testing/README.md

```diff
@@ -0,0 +1,540 @@
+for additional test scenarios. The logic here is that we don't want
+to send Ether to the target address on L2 because we might risk
+withdrawing more than we deposit
+
+```
+cast send --rpc-url $l1_rpc_url --private-key $private_key $pol_address 'mint(address,uint256)' $target_address 10000000000000000000000
+cast send --rpc-url $l1_rpc_url --private-key $private_key $pol_address 'approve(address,uint256)' $bridge_address 10000000000000000000000
+
+cast send --rpc-url $l1_rpc_url --private-key $private_key $gas_token_address 'mint(address,uint256)' $target_address 10000000000000000000000
+cast send --rpc-url $l1_rpc_url --private-key $private_key $gas_token_address 'approve(address,uint256)' $bridge_address 10000000000000000000000
+```
+
+We're going to do a little matrix of L1 to L2 bridges here. The idea
+is to mix bridges across a few scenarios
+- Native vs ERC20
+- Bridge Message vs Bridge Asset
+- GER Updating
+
+```
+dry_run=false
+for token_addr in $(cast az) $pol_address $gas_token_address ; do
+    for destination_network in 1 2 3 ; do
+        for ger_update in "false" "true" ; do
+            permit_data="0x"
+            value="1000000000000000000"
+            if [[ $token_addr == $(cast az) ]]; then
+                permit_data="0x$(cat /dev/random | xxd -p | tr -d "\n" | head -c $((RANDOM & ~1)))"
+            fi
+            polycli ulxly bridge asset \
+                    --private-key $private_key \
+                    --value $value \
+                    --bridge-address $bridge_address \
+                    --destination-network $destination_network \
+                    --destination-address $target_address \
+                    --call-data $permit_data \
+                    --rpc-url $l1_rpc_url \
+                    --token-address $token_addr \
+                    --force-update-root=$ger_update \
+                    --dry-run=$dry_run
+            polycli ulxly bridge message \
+                    --private-key $private_key \
+                    --value $value \
+                    --bridge-address $bridge_address \
+                    --destination-network $destination_network \
+                    --destination-address $target_address \
+                    --call-data $permit_data \
+                    --rpc-url $l1_rpc_url \
+                    --force-update-root=$ger_update \
+                    --dry-run=$dry_run
+        done
+    done
+done
+```
+
+At this point, we should be able to see our bridges in the bridge
+service for both chains. Recently, with this setup there has been an
+issue with network id detection in the bridge service
+
+```
+curl -s $l2_pp1b_url/bridges/$target_address | jq '.'
+curl -s $l2_pp2b_url/bridges/$target_address | jq '.'
+curl -s $l2_fepb_url/bridges/$target_address | jq '.'
+```
+
+Some of the bridges will be claimed already, but some will need to
+be claimed manually. Most of the ones that need to be claimed
```

Comment:
Does this happen because the claim tx manager automatically claims asset bridge deposits and not messages?

Reply:
That's right.